[DOCS] Docker and GPU update for 23.0 (#17851)
openvinotoolkit/openvino: docs/install_guides/configurations-for-intel-gpu.md

@sphinxdirective

.. _gpu guide:

To use the OpenVINO™ GPU plug-in and transfer the inference to the graphics of the Intel® processor (GPU), the Intel® graphics driver must be properly configured on the system.

Linux
#####

To use a GPU device for OpenVINO inference, you must meet the following prerequisites:

- use a supported Linux kernel as per the `documentation <https://dgpu-docs.intel.com/driver/kernel-driver-types.html>`__
- install GPU Runtime drivers: `The Intel(R) Graphics Compute Runtime for oneAPI Level Zero and OpenCL(TM) Driver <https://github.com/intel/compute-runtime/releases/latest>`__
- install `Intel Graphics Memory Management Library <https://github.com/intel/gmmlib>`__ and `Intel® Graphics Compiler for OpenCL™ <https://github.com/intel/intel-graphics-compiler>`__
- install `OpenCL ICD loader package <https://github.com/KhronosGroup/OpenCL-ICD-Loader>`__

Depending on your operating system, there may be different methods to install the above packages. Below are the instructions on how to install the packages on supported Linux distributions.

Ubuntu 22.04 LTS

> ++++++++++++++++

Reviewer: should we turn it into tabs? So, I can select between Ubuntu 22, Ubuntu 20 or UBI 8

Download and install the `deb` packages published `here <https://github.com/intel/compute-runtime/releases/latest>`__ and install the apt package `ocl-icd-libopencl1` with the OpenCl ICD loader.

Alternatively, add the apt repository (as described here `here <https://dgpu-docs.intel.com/driver/installation.html>`__). For example, for an ARC device, use the command:

.. code-block:: sh
   
   apt-get update && apt-get install -y gpg gpg-agent curl
   curl https://repositories.intel.com/graphics/intel-graphics.key | \
     gpg --dearmor --output /usr/share/keyrings/intel-graphics.gpg

>    echo 'deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/graphics/ubuntu jammy arc' | \

Reviewer: installation of GPU drivers is external thing with regards to OpenVINO.
We don't want to show exact packages which need to be installed. Supposed that users should configure GPU before running OpenVINO.

Reviewer 2: @ilya-lavrenov relying purely on external GPU documentation has two problems: 
- the documentation include dependencies not needed by openvino like GUI applications dependencies and media decoding accelerators.
- there are packages which are not mentioned as required like ocl-icd
I think be adding a guide which is easy to follow and more specific for OV will save us and the users time on troubleshooting.

Reviewer: `ocl-icd` is a direct dependency of Intel GPU and we have to add it to our `install_openvino_dependencies.sh` script.

> the documentation include dependencies not needed by openvino like GUI applications dependencies and media decoding accelerators.

This is not OpenVINO problem, we are not going to rewrite documentation of our dependencies.

Reviewer: Added ICD loader as a dependency https://github.com/openvinotoolkit/openvino/pull/18123

# Configurations for Intel® Processor Graphics (GPU) with OpenVINO™ {#openvino_docs_install_guides_configurations_for_intel_gpu}

@sphinxdirective

.. _gpu guide:

To use the OpenVINO™ GPU plug-in and transfer the inference to the graphics of the Intel® processor (GPU), the Intel® graphics driver must be properly configured on the system.

Linux
#####

To use a GPU device for OpenVINO inference, you must meet the following prerequisites:

- use a supported Linux kernel as per the `documentation <https://dgpu-docs.intel.com/driver/kernel-driver-types.html>`__
- install GPU Runtime drivers: `The Intel(R) Graphics Compute Runtime for oneAPI Level Zero and OpenCL(TM) Driver <https://github.com/intel/compute-runtime/releases/latest>`__
- install `Intel Graphics Memory Management Library <https://github.com/intel/gmmlib>`__ and `Intel® Graphics Compiler for OpenCL™ <https://github.com/intel/intel-graphics-compiler>`__

> - install `OpenCl ICD loader package <https://github.com/KhronosGroup/OpenCL-ICD-Loader>`__

Reviewer 3: ```suggestion
- install `OpenCL ICD loader package <https://github.com/KhronosGroup/OpenCL-ICD-Loader>`__
```

> Depending on your operating system, there may be different methods to install the above packages. Below are the instructions on how to install the packages, broken down by the operating system you want to install on.

Reviewer 3: ```suggestion
Depending on your operating system, there may be different methods to install the above packages. Below are the instructions on how to install the packages on supported Linux distributions.
```

.. code-block:: sh
   
   apt-get update && apt-get install -y --no-install-recommends curl gpg gpg-agent && \
   curl https://repositories.intel.com/graphics/intel-graphics.key | gpg --dearmor --output /usr/share/keyrings/intel-graphics.gpg && \
   echo 'deb [arch=amd64 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/graphics/ubuntu focal-legacy main' | tee  /etc/apt/sources.list.d/intel.gpu.focal.list && \
   apt-get update
   apt-get update && apt-get install -y --no-install-recommends intel-opencl-icd intel-level-zero-gpu level-zero

Alternatively, download older `deb` version from `here <https://github.com/intel/compute-runtime/releases>`__. Note that older driver version might not include some of the bug fixes and might be not supported on some latest platforms. Check the supported hardware for the versions you are installing.

RedHat UBI 8
++++++++++++

Follow the `instructions <https://dgpu-docs.intel.com/driver/installation.html#red-hat-enterprise-linux-8-6>`__ to add Yum repository.

Install following packages: 

* `intel-opencl`,
* `level-zero`,
* `intel-level-zero-gpu` ,
* `intel-igc-core`, 
* `intel-igc-cm`, 
* `intel-gmmlib`,

> * `intel-ocloc`.

Reviewer 2: It should be `yum install intel-opencl level-zero intel-level-zero-gpu intel-igc-core intel-igc-cm intel-gmmlib intel-ocloc

> * `intel-ocloc`.

Author: The change has been implemented in https://github.com/openvinotoolkit/openvino/pull/17851/commits/327c0951884a6d29a4c50c1750a15e6ca1e25c76
